File: sensor-graph/src/androidTest/java/com/example/hellodigitalis/sensorgraph/ScreenshotTest.java
package com.example.hellodigitalis.sensorgraph;

import androidx.test.ext.junit.runners.AndroidJUnit4;

import com.example.hellodigitalis.screenshottest.ScreenshotTestRule;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4.class)
public class ScreenshotTest {
    @Rule
    public ScreenshotTestRule rule = new ScreenshotTestRule(
            "com.example.hellodigitalis.sensorgraph/com.example.hellodigitalis.sensorgraph.AccelerometerGraphActivity",
            5000,
            0.10f
    );

    @Test
    public void screenshotMatchesReference() {
        rule.assertMatchesReference("screenshot_default.png");
    }
}

File: teapots-textured/src/androidTest/java/com/example/hellodigitalis/teapotstextured/ScreenshotTest.kt
package com.example.hellodigitalis.teapotstextured

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.example.hellodigitalis.screenshottest.ScreenshotTestRule
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class ScreenshotTest {
    @get:Rule
    val rule = ScreenshotTestRule(
        "com.example.hellodigitalis.teapotstextured/com.sample.texturedteapot.TeapotNativeActivity",
        5000,
        0.05f
    )

    @Test
    fun screenshotMatchesReference() {
        rule.assertMatchesReference("screenshot_default.png")
    }
}
